specification/elasticsan/resource-manager/Microsoft.ElasticSan/preview/2024-07-01-preview/elasticsan.json

```diff
@@ -2840,11 +2814,18 @@
       "type": "object",
       "description": "Response for Delete Retention Policy object",
       "properties": {
-        "enabled": {
-          "type": "boolean",
-          "description": "The flag to enable/disable retention policy."
+        "policyState": {
```

Comment:
In Storage RP, all DeleteRetentionPolicy are defined like below. 

Do we want to use different name for same meaning in ElasticSan?
https://github.com/Azure/azure-rest-api-specs/blob/9a73f1c2121fc835bf97ccf586df8a0019a130c8/specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/common.json#L83

```
    "DeleteRetentionPolicy": {
      "properties": {
        "enabled": {
          "type": "boolean",
          "description": "Indicates whether DeleteRetentionPolicy is enabled."
        },
        "days": {
          "type": "integer",
          "minimum": 1,
          "maximum": 365,
          "description": "Indicates the number of days that the deleted item should be retained. The minimum specified value can be 1 and the maximum value can be 365."
        },
        "allowPermanentDelete": {
          "type": "boolean",
          "description": "This property when set to true allows deletion of the soft deleted blob versions and snapshots. This property cannot be used blob restore policy. This property only applies to blob service and does not apply to containers or file share."
        }
      },
      "description": "The service properties for soft delete."
    }
    ```

Reply:
Please give ARM reviewer the context that the before name is aligned with storage, after change it's not aligned.

Reply:
@blueww , I think the suggestions with respect to having "enum" instead of boolean and a more descriptive name for "retentionPeriodInDays" make sense to have (and make it more readable/usable as well).

cc @mentat9

Reply:
Just to make sure it's considered that with the current swagger, same setting use different name/type in Storage RP and ElaticSan RP.